pkgs/tools/filesystems/dduper/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv, fetchFromGitHub, python3Packages, btrfs-progs }:
+
+let
+  our-btrfs-progs = btrfs-progs.override {
+    patches = [ patch ];
+  };
+
+  version = "0.04";
+
+  src = fetchFromGitHub {
+    owner = "Lakshmipathi";
+    repo = "dduper";
+    rev = "v${version}";
+    sha256 = "09ncdawxkffldadqhfblqlkdl05q2qmywxyg6p61fv3dr2f2v5wm";
+  };
+
+  patch = stdenv.mkDerivation {
+    name = "btrfs-csum.patch";
+
+    inherit src;
+
+    phases = [ "unpackPhase" "installPhase" ];
+
+    installPhase = ''
+      cp patch/btrfs-progs-v5.6.1/0001-Print-csum-for-a-given-file-on-stdout.patch $out
+    '';
+  };
+in
+python3Packages.buildPythonPackage rec {
+  pname = "dduper";
+
+  inherit src version;
+
+  buildInputs = [ our-btrfs-progs ];
+
+  pythonPath = [ python3Packages.python
+    python3Packages.numpy
+    python3Packages.ptable ];
+
+  prePatch = ''
+    sed -i 's,/usr/sbin/btrfs\.static,'${our-btrfs-progs}/bin/btrfs, dduper
```

Comment:
sed is not ideal because it silently fails.

Reply:
good point. What would you use?

Reply:
you can either use `substituteInPlace` (which prints a warning if nothing is replace) or use a full blown patch (which fails loudly to apply in case upstream changes the corresponding lines) like in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/graphics/gimp/default.nix#L67-L70